Revit_Core_Engine/Create/View/Elevation.cs

```diff
@@ -0,0 +1,187 @@
+            var angle = -markerDirection.AngleTo(XYZ.BasisX);
+            var axis = Line.CreateBound(elevationMarkerLocation, elevationMarkerLocation + XYZ.BasisZ);
+
+            elevationMarker = ElevationMarker(elevationMarkerLocation, referenceViewPlan);
+            var elevationView = elevationMarker.CreateElevation(doc, referenceViewPlan.Id, 0);
+
+            ElementTransformUtils.RotateElement(doc, elevationMarker.Id, axis, angle);
+
+            elevationView.ModifyElevation(elevationName, elevationLine, depth, height, offset, viewTemplateId, cropRegionVisible, annotationCrop);
+
+            return elevationView;
+        }
+
+        /***************************************************/
+
+        [Description("Creates elevation views referenced do elevation marker in the current Revit file.")]
+        [Input("elevationMarker", "Elevation marker to which elevation views will be referenced.")]
+        [Input("elevationName", "Name of the elevation view.")]
+        [Input("referenceViewPlan", "ViewPlan that elevation marker and elevation views will be referenced to.")]
+        [Input("elevationIndex", "The index on the Elevation Marker where the new elevation will be placed.")]
+        [Input("bottomLine", "Location line of the elevation view and the bottom line for the crop region shape.")]
+        [Input("depth", "Depth of the elevation view.")]
+        [Input("height", "Height of the elevation view.")]
+        [Input("viewTemplateId", "Optional, the View Template Id to be applied in the view.")]
+        [Input("cropRegionVisible", "True if the crop region should be visible.")]
+        [Input("annotationCrop", "True if the annotation crop should be visible.")]
+        [Output("elevations", "The list of created elevations.")]
+        public static ViewSection ElevationView(this ElevationMarker elevationMarker, string elevationName, View referenceViewPlan, int elevationIndex, Line bottomLine, double depth, double height, double offset, ElementId viewTemplateId = null, bool cropRegionVisible = false, bool annotationCrop = false)
+        {
+            Document doc = elevationMarker.Document;
+
+            //create ViewSection
+            ViewSection newElevation = elevationMarker.CreateElevation(doc, referenceViewPlan.Id, elevationIndex);
+            newElevation.ModifyElevation(elevationName, bottomLine, depth, height, offset, viewTemplateId, cropRegionVisible, annotationCrop);
+
+            return newElevation;
+        }
+
+        /***************************************************/
+        /****              Private Methods              ****/
+        /***************************************************/
+
+        [Description("Creates elevation marker in the current Revit file.")]
+        private static ElevationMarker ElevationMarker(this XYZ elevationMarkerLocation, View referenceViewPlan)
```

Comment:
Wouldn't it be worth for this method to be public?

Reply:
Done 👍